### PR TITLE
Update documentation for deploying helium in AKS using Istio

### LINKED ISSUE #143: Helium demo in a box webcast
- Work with tech domain to deliver and record a webcast on Helium demo-in-a-box

### PR DESCRIPTION
**Pull request by joaquinrz**
# Type of PR

- [x] Documentation changes
- [x] Code changes

## Purpose of PR
Updating the helium installation in AKS using Istio instead of Linkerd + NGINX and also using Keda for pod autoscaling.

## Validation
- [X] Update documentation or issue referenced above

### CODE REVIEW

**Review by bartr — CHANGES_REQUESTED** [review]
Blocking for discussion at stand up

File: docs/aks/README.md [hunk 1]
```diff
@@ -492,9 +458,11 @@ Replace the values in the `AKS_Dashboard.json` file surrounded by `%%` with the
 ```bash
 
 cd $REPO_ROOT/docs/aks/demo
-sed -i "s/%%SUBSCRIPTION_GUID%%/${He_Sub}/g" AKS_Dashboard.json && \
+sed -i "s/%%SUBSCRIPTION_GUID%%/$(eval $He_Sub)/g" AKS_Dashboard.json && \
```
Comment (AAkindele) [inline docs/aks/README.md:461]: [blocker] I ran into an error here with eval. He_Sub is the value of the subscription and not the command string. So the eval fails.

**Review by AAkindele — CHANGES_REQUESTED** [review]
Found 1 problem with the dashboard command. But, besides that, everything else looks good.

File: docs/aks/README.md [hunk 2]
```diff
@@ -21,7 +20,7 @@ Project Helium is a reusable Advocated Pattern (AdPat). The focus was originally
 ### Prerequisites
 
 - Azure subscription with permissions to create:
-  - Resource Groups, Service Principals, Keyvault, Cosmos DB, AKS, Azure Container Registry, Azure Monitor
+  - Resource Groups, Service Principals, Keyvault, Cosmos DB, AKS, Azure Monitor
```
Comment (bartr) [inline docs/aks/README.md:23]: Per design review, we are going to leave ACR and Az Monitor in helium

File: docs/aks/README.md [hunk 3]
```diff
@@ -70,7 +69,7 @@ az account set -s {subscription name or Id}
 
 ```
 
-This demo will create resource groups, a Cosmos DB instance, Key Vault, Azure Container Registry, and Azure App Service.
+This demo will create resource groups, a Cosmos DB instance, Key Vault, and Azure App Service.
```
Comment (bartr) [inline docs/aks/README.md:72]: per design review - leave ACR

File: docs/aks/README.md [hunk 4]
```diff
@@ -97,8 +96,7 @@ nslookup ${He_Name}.azurecr.io
 >
 > If you use an existing resource group, please make sure to apply resource locks to avoid accidentally deleting resources
 
-- You will create 3 resource groups
-  - One for ACR
+- You will create 2 resource groups
```
Comment (bartr) [inline docs/aks/README.md:99]: leave ACR

File: docs/aks/README.md [hunk 5]
```diff
@@ -112,7 +110,6 @@ export He_Sub=$(az account show --query id -o tsv)
 
 # resource group names
 export Imdb_Name=$He_Name
-export He_ACR_RG=${He_Name}-rg-acr
```
Comment (bartr) [inline docs/aks/README.md:115]: leave ACR

File: docs/aks/README.md [hunk 6]
```diff
@@ -125,7 +122,6 @@ export Imdb_RW_Key='az cosmosdb keys list -n $Imdb_Name -g $Imdb_RG --query prim
 
 # create the resource groups
 az group create -n $He_App_RG -l $He_Location
-az group create -n $He_ACR_RG -l $He_Location
```
Comment (bartr) [inline docs/aks/README.md:128]: leave ACR

File: docs/aks/README.md [hunk 7]
```diff
@@ -204,17 +200,6 @@ export He_AppInsights_Key='az keyvault secret show -o tsv --query value --vault-
 
 ```
 
-#### Setup Azure Container Registry
```
Comment (bartr) [inline docs/aks/README.md:207]: leave ACR

File: docs/aks/README.md [hunk 8]
```diff
@@ -204,17 +200,6 @@ export He_AppInsights_Key='az keyvault secret show -o tsv --query value --vault-
...
-
-- Create the Container Registry with admin access `disabled`
-
-```bash
-
-# create the ACR
-az acr create --sku Standard --admin-enabled false -g $He_ACR_RG -n $He_Name
-
-```
-
```
Comment (bartr) [inline docs/aks/README.md:217]: leave ACR

File: docs/AppService-dotnetcore.md [hunk 9]
```diff
@@ -53,11 +56,11 @@ Run the Validation Test
 
 ```bash
 
-# run the tests in the container
-docker run -it --rm retaildevcrew/webvalidate --server https://${He_Name}.azurewebsites.net --files helium.json
+# run the tests in a container
+docker run -it --rm retaildevcrew/webvalidate --server $He_App_Endpoint --base-url https://raw.githubusercontent.com/retaildevcrews/helium/main/TestFiles/ --files baseline.json
```
Comment (sivamu) [inline docs/AppService-dotnetcore.md:60]: I wasn't able to get helium.json to work, hence I changed this to use the baseurl instead.

**Review by AAkindele — APPROVED** [review]

File: docs/AppObservability.md [hunk 10]
```diff
@@ -158,7 +158,7 @@ curl -s https://raw.githubusercontent.com/retaildevcrews/helium/main/docs/dashbo
 sed -i "s/%%SUBSCRIPTION_GUID%%/$(eval $He_Sub)/g" Helium_Dashboard.json
```
Comment (jofultz) [inline docs/AppObservability.md:158]: Are these valid for an AKS deployment?  The previous dashboards were based on App Services reporting to App Insights.

File: docs/AppService-dotnetcore.md [hunk 11]
```diff
@@ -33,6 +33,9 @@ az webapp deployment source delete -g $He_App_RG -n $He_Name
 # this step usually takes 3-5 minutes
 az webapp deployment source config -g $He_App_RG -n $He_Name -u https://github.com/retaildevcrews/helium-csharp --branch main --manual-integration
 
+# set the app endpoint
+export He_App_Endpoint=https://${He_Name}.azurewebsites.net
```
Comment (jofultz) [inline docs/AppService-dotnetcore.md:37]: Is this valid?  I think [name].azurewebsites.net is only App Services based sites.

File: docs/AppService.md [hunk 12]
```diff
@@ -33,6 +33,9 @@ az webapp config appsettings set --settings KEYVAULT_NAME=$He_Name -g $He_App_RG
 # this will send stdout and stderr to the logs
 az webapp log config --docker-container-logging filesystem -g $He_App_RG -n $He_Name
 
+# set the app endpoint
+export He_App_Endpoint=https://${He_Name}.azurewebsites.net
+
```
Comment (jofultz) [inline docs/AppService.md:38]: Same questions and notes regarding App Service versus AKS.

File: docs/AppService.md [hunk 13]
```diff
@@ -63,11 +66,11 @@ Run the Validation Test
...
-docker run -it --rm retaildevcrew/webvalidate --server https://${He_Name}.azurewebsites.net --files helium.json
+# run the tests in a container
+docker run -it --rm retaildevcrew/webvalidate --server $He_App_Endpoint --base-url https://raw.githubusercontent.com/retaildevcrews/helium/main/TestFiles/ --files baseline.json
 
 ```
 
 ## Observability
 
-See [App Service Observability](AppServiceObservability.md) for details
+See [App Observability](AppObservability.md) for details
```
Comment (jofultz) [inline docs/AppService.md:76]: Might want to leave it as AppServiceObservability since the observability for the AKS deployment would be different.

Comment (sivamu) [reply docs/AppService.md:76]: I'll do some more testing with the dashboards and verify whether the App Service dashboard can work with AKS.

**Review by bartr — APPROVED** [review]

**Pull request merged**